mlx/backend/metal/kernels/ternary.metal

```diff
@@ -99,103 +102,107 @@ template <typename T, typename Op>
+      constant const size_t c_strides[dims],             \
+      uint3 index [[thread_position_in_grid]],           \
+      uint3 grid_dim [[threads_per_grid]]);
 
+#define instantiate_ternary_g_nd(name, type, op)    \
+  template [[host_name(name "_1")]] [[kernel]] void \
+  ternary_op_g_nd1<type, op>(                       \
+      device const bool* a,                         \
+      device const type* b,                         \
+      device const type* c,                         \
+      device type* d,                               \
+      constant const size_t& a_strides,             \
+      constant const size_t& b_strides,             \
+      constant const size_t& c_strides,             \
+      uint index [[thread_position_in_grid]]);      \
+  template [[host_name(name "_2")]] [[kernel]] void \
+  ternary_op_g_nd2<type, op>(                       \
+      device const bool* a,                         \
+      device const type* b,                         \
+      device const type* c,                         \
+      device type* d,                               \
+      constant const size_t a_strides[2],           \
+      constant const size_t b_strides[2],           \
+      constant const size_t c_strides[2],           \
+      uint2 index [[thread_position_in_grid]],      \
+      uint2 grid_dim [[threads_per_grid]]);         \
+  template [[host_name(name "_3")]] [[kernel]] void \
+  ternary_op_g_nd3<type, op>(                       \
+      device const bool* a,                         \
+      device const type* b,                         \
+      device const type* c,                         \
+      device type* d,                               \
+      constant const size_t a_strides[3],           \
+      constant const size_t b_strides[3],           \
+      constant const size_t c_strides[3],           \
+      uint3 index [[thread_position_in_grid]],      \
+      uint3 grid_dim [[threads_per_grid]]);         \
+  instantiate_ternary_g_dim(name, type, op, 4)      \
+      instantiate_ternary_g_dim(name, type, op, 5)
+
+// clang-format off
 #define instantiate_ternary_all(name, tname, type, op) \
-  instantiate_ternary_v("v" #name #tname, type, op) \
-  instantiate_ternary_g("g" #name #tname, type, op) \
-  instantiate_ternary_g_nd("g" #name #tname, type, op) \
-
-#define instantiate_ternary_types(name, op) \
-  instantiate_ternary_all(name, bool_, bool, op) \
-  instantiate_ternary_all(name, uint8, uint8_t, op) \
-  instantiate_ternary_all(name, uint16, uint16_t, op) \
-  instantiate_ternary_all(name, uint32, uint32_t, op) \
-  instantiate_ternary_all(name, uint64, uint64_t, op) \
-  instantiate_ternary_all(name, int8, int8_t, op) \
-  instantiate_ternary_all(name, int16, int16_t, op) \
-  instantiate_ternary_all(name, int32, int32_t, op) \
-  instantiate_ternary_all(name, int64, int64_t, op) \
-  instantiate_ternary_all(name, float16, half, op) \
+  instantiate_ternary_g("g" #name #tname, type, op)    \
```

Comment:
Looks like you dropped a line here?

Reply:
Oh I think these errors raised due to me resolving merge conflicts. Sorry

Reply:
Any idea why this might happen

```
  test_compile_broadcast_only (test_compile.TestCompile) ... libc++abi: terminating due to uncaught exception of type std::runtime_error: [metal::Device] Unable to load build metal library from source
program_source:13:17: error: use of undeclared identifier 'Add'
  float tmp_B = Add()(tmp_C, tmp_C);
                ^
program_source:27:17: error: use of undeclared identifier 'Add'
  float tmp_B = Add()(tmp_C, tmp_C);
                ^
program_source:41:17: error: use of undeclared identifier 'Add'
  float tmp_B = Add()(tmp_C, tmp_C);
                ^
program_source:55:17: error: use of undeclared identifier 'Add'
  float tmp_B = Add()(tmp_C, tmp_C);
                ^
program_source:69:17: error: use of undeclared identifier 'Add'
  float tmp_B = Add()(tmp_C, tmp_C);
                ^
program_source:83:17: error: use of undeclared identifier 'Add'
  float tmp_B = Add()(tmp_C, tmp_C);
                ^
program_source:97:17: error: use of undeclared identifier 'Add'
  float tmp_B = Add()(tmp_C, tmp_C);
                ^
program_source:111:17: error: use of undeclared identifier 'Add'
  float tmp_B = Add()(tmp_C, tmp_C);
                ^
program_source:126:17: error: use of undeclared identifier 'Add'
  float tmp_B = Add()(tmp_C, tmp_C);
                ^


fish: Job 1, 'LOW_MEMORY=1 DEVICE=gpu METAL_D…' terminated by signal SIGABRT (Abort)
```

Reply:
> Any idea why this might happen
> 
> ```
>   test_compile_broadcast_only (test_compile.TestCompile) ... libc++abi: terminating due to uncaught exception of type std::runtime_error: [metal::Device] Unable to load build metal library from source
> program_source:13:17: error: use of undeclared identifier 'Add'
>   float tmp_B = Add()(tmp_C, tmp_C);
>                 ^
> program_source:27:17: error: use of undeclared identifier 'Add'
>   float tmp_B = Add()(tmp_C, tmp_C);
>                 ^
> program_source:41:17: error: use of undeclared identifier 'Add'
>   float tmp_B = Add()(tmp_C, tmp_C);
>                 ^
> program_source:55:17: error: use of undeclared identifier 'Add'
>   float tmp_B = Add()(tmp_C, tmp_C);
>                 ^
> program_source:69:17: error: use of undeclared identifier 'Add'
>   float tmp_B = Add()(tmp_C, tmp_C);
>                 ^
> program_source:83:17: error: use of undeclared identifier 'Add'
>   float tmp_B = Add()(tmp_C, tmp_C);
>                 ^
> program_source:97:17: error: use of undeclared identifier 'Add'
>   float tmp_B = Add()(tmp_C, tmp_C);
>                 ^
> program_source:111:17: error: use of undeclared identifier 'Add'
>   float tmp_B = Add()(tmp_C, tmp_C);
>                 ^
> program_source:126:17: error: use of undeclared identifier 'Add'
>   float tmp_B = Add()(tmp_C, tmp_C);
>                 ^
> 
> 
> fish: Job 1, 'LOW_MEMORY=1 DEVICE=gpu METAL_D…' terminated by signal SIGABRT (Abort)
> ```

@awni 
I am getting this error when running the test from the main branch too.